From 63ddc9bb61e011444ffa7e5b665153ea1d0eb829 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Tue, 18 Aug 2026 21:29:48 +0200
Subject: partial Haiku build fix

Use gnu++17 as the C++ standard instead of c++17 for netdb error codes
used by Boost ASIO.

Use gnu99 C standard for usrsctp and arc4random like BSD and Apple.

diff --git a/Jamfile b/Jamfile
index 9abd8b5..e7cb141 100644
--- a/Jamfile
+++ b/Jamfile
@@ -1052,6 +1052,7 @@ lib torrent
 	: # default build
 	<threading>multi
 	<cxxstd>17
+	<cxxstd-dialect>gnu
 	<c++-template-depth>512
 
 	: # usage requirements
diff --git a/deps/libdatachannel/deps/usrsctp/CMakeLists.txt b/deps/libdatachannel/deps/usrsctp/CMakeLists.txt
index c3467f2..35ef6b5 100644
--- a/deps/libdatachannel/deps/usrsctp/CMakeLists.txt
+++ b/deps/libdatachannel/deps/usrsctp/CMakeLists.txt
@@ -200,7 +200,7 @@ if (C_COMPILER_IS_MSVC_LIKE)
 	endif ()
 # SETTINGS FOR UNIX COMPILER
 elseif (CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID MATCHES "AppleClang" OR CMAKE_C_COMPILER_ID MATCHES "GNU")
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -pedantic -Wall -Wextra")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pedantic -Wall -Wextra")
 
 	check_c_compiler_flag(-Wfloat-equal has_wfloat_equal)
 	if (has_wfloat_equal)
diff --git a/deps/libdatachannel/deps/usrsctp/usrsctplib/user_environment.c b/deps/libdatachannel/deps/usrsctp/usrsctplib/user_environment.c
index ea52f0a..2d41325 100755
--- a/deps/libdatachannel/deps/usrsctp/usrsctplib/user_environment.c
+++ b/deps/libdatachannel/deps/usrsctp/usrsctplib/user_environment.c
@@ -120,7 +120,7 @@ finish_random(void)
 {
 	return;
 }
-#elif defined(__FreeBSD__) || defined(__DragonFly__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__) || defined(__Bitrig__)
+#elif defined(__FreeBSD__) || defined(__DragonFly__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__) || defined(__Bitrig__) || defined(__HAIKU__)
 #include <stdlib.h>
 
 void
-- 
2.54.0

